### DISCUSSION TITLE
prefix url

### DISCUSSION THREAD

**Discussion by danielehrhardt** [ngneat/transloco#491] [General]
Hi,

I wan't to ask how you would realize prefix language routing. 

What do I mean with that? 
domain.com/en/page1 
domain.com/de/page2

I created a APP INITIALIZER Service and check if there is the URL and then set the language to that URL. 

The user can overwrite manually the language this value is then saved to the local storage. 

In the angular router I do the following:

:lang/page1 

In Vue the translation Plugin can handle the URL Language prefix by itself. For ngx-translate there is a Plugin that can handle that. 

Now I did all by myself. 

Is there a better way? 

Best
Daniel

**Comment by danielehrhardt** [comment]
Something like that:
https://i18n.nuxtjs.org/browser-language-detection/

**Comment by danielehrhardt** [comment]
https://github.com/ngneat/transloco/issues/28

**Comment by shaharkazaz** [comment]
@danielehrhardt we currently don't have any official plugin for that, so you can implement it in any way that best suits your needs 🙂
